src/basic/fs-util.c

```diff
@@ -195,13 +195,13 @@ int readlink_and_make_absolute(const char *p, char **r) {
         return 0;
 }
 
-int chmod_and_chown(const char *path, mode_t mode, uid_t uid, gid_t gid) {
+int chmod_and_chown_at(int dir_fd, const char *path, mode_t mode, uid_t uid, gid_t gid) {
         _cleanup_close_ int fd = -1;
 
         assert(path);
 
-        fd = open(path, O_PATH|O_CLOEXEC|O_NOFOLLOW); /* Let's acquire an O_PATH fd, as precaution to change
-                                                       * mode/owner on the same file */
+        fd = openat(dir_fd, path, O_PATH|O_CLOEXEC|O_NOFOLLOW); /* Let's acquire an O_PATH fd, as precaution
+                                                                 * to change mode/owner on the same file */
         if (fd < 0)
                 return -errno;
 
```

Comment:
hmm, if path is specified as NULL, why not shortcut this to fchmod_and_chown() and pass dir_fd? sounds more versatile that way.